editor/import/audio_stream_import_settings.cpp

```diff
@@ -0,0 +1,648 @@
+	get_ok_button()->set_text(TTR("Reimport"));
+	get_cancel_button()->set_text(TTR("Close"));
+
+	VBoxContainer *main_vbox = memnew(VBoxContainer);
+	add_child(main_vbox);
+
+	HBoxContainer *loop_hb = memnew(HBoxContainer);
+	loop_hb->add_theme_constant_override("separation", 4 * EDSCALE);
+	loop = memnew(CheckBox);
+	loop->set_text(TTR("Enable"));
+	loop->set_tooltip(TTR("Enable looping."));
+	loop->connect("toggled", callable_mp(this, &AudioStreamImportSettings::_settings_changed).unbind(1));
+	loop_hb->add_child(loop);
+	loop_hb->add_spacer();
+	loop_hb->add_child(memnew(Label(TTR("Offset (in sec):"))));
+	loop_offset = memnew(SpinBox);
+	loop_offset->set_max(10000);
+	loop_offset->set_step(0.001);
+	loop_offset->set_tooltip(TTR("Loop offset (from beginning). Note that if BPM is set, this setting will be ignored."));
+	loop_offset->connect("value_changed", callable_mp(this, &AudioStreamImportSettings::_settings_changed).unbind(1));
+	loop_hb->add_child(loop_offset);
+	main_vbox->add_margin_child(TTR("Loop:"), loop_hb);
+
+	HBoxContainer *interactive_hb = memnew(HBoxContainer);
+	interactive_hb->add_theme_constant_override("separation", 4 * EDSCALE);
+	bpm_enabled = memnew(CheckBox);
+	bpm_enabled->set_text((TTR("BPM:")));
+	bpm_enabled->connect("toggled", callable_mp(this, &AudioStreamImportSettings::_settings_changed).unbind(1));
+	interactive_hb->add_child(bpm_enabled);
+	bpm_edit = memnew(SpinBox);
+	bpm_edit->set_max(400);
+	bpm_edit->set_step(0.01);
+	bpm_edit->set_tooltip(TTR("Configure the Beats Per Measure (tempo) used for the interactive streams\n (AudioStreamTransitioner and AudioStreamPlaylist).\nIf zero, the one from the parent stream will be used."));
+	bpm_edit->connect("value_changed", callable_mp(this, &AudioStreamImportSettings::_settings_changed).unbind(1));
+	interactive_hb->add_child(bpm_edit);
+	interactive_hb->add_spacer();
+	bar_beats_label = memnew(Label(TTR("Beats/Bar:")));
+	interactive_hb->add_child(bar_beats_label);
+	bar_beats_edit = memnew(SpinBox);
+	bar_beats_edit->set_tooltip(TTR("Configure the Beats Per Bar. This used for music-aware transitions between AudioStreams."));
+	bar_beats_edit->set_min(2);
+	bar_beats_edit->set_max(32);
+	bar_beats_edit->connect("value_changed", callable_mp(this, &AudioStreamImportSettings::_settings_changed).unbind(1));
+	interactive_hb->add_child(bar_beats_edit);
+	interactive_hb->add_spacer();
+	beats_enabled = memnew(CheckBox);
+	beats_enabled->set_text(TTR("Length (in beats):"));
+	beats_enabled->connect("toggled", callable_mp(this, &AudioStreamImportSettings::_settings_changed).unbind(1));
+	interactive_hb->add_child(beats_enabled);
+	beats_edit = memnew(SpinBox);
+	beats_edit->set_tooltip(TTR("Configure the Beats Per Measure (tempo) used for the interactive streams\n (AudioStreamTransitioner and AudioStreamPlaylist).\bIf zero, it will be autodetected from the length.\nIt is recommended to set this value to ensure looping works properly."));
+	beats_edit->set_max(99999);
```

Comment:
```suggestion
	beats_enabled = memnew(CheckBox);
	beats_enabled->set_text(TTR("Length:"));
	beats_enabled->connect("toggled", callable_mp(this, &AudioStreamImportSettings::_settings_changed).unbind(1));
	interactive_hb->add_child(beats_enabled);
	beats_edit = memnew(SpinBox);
	beats_edit->set_tooltip(TTR("Configure the Beats Per Measure (tempo) used for the interactive streams\n (AudioStreamTransitioner and AudioStreamPlaylist).\bIf zero, it will be autodetected from the length.\nIt is recommended to set this value to ensure looping works properly."));
	beats_edit->set_max(99999);
	beats_edit->set_suffix("beats");
```